lib/combineNodes.js

```diff
@@ -0,0 +1,310 @@
+'use strict';
+var arrayEqual = require('array-equal');
+var Cesium = require('cesium');
+var clone = Cesium.clone;
+var defined = Cesium.defined;
+var defaultValue = Cesium.defaultValue;
+var Matrix4 = Cesium.Matrix4;
+var Cartesian3 = Cesium.Cartesian3;
+var combineMeshes = require('./combineMeshes');
+var combinePrimitives = require('./combinePrimitives');
+module.exports = combineNodes;
+
+//Combines nodes which are not targeted for animation and share the same material. 
+//Should be run after the combineMeshes and combinePrimitives stages, giving nodes one mesh each.
+function combineNodes(gltf, preserveCameras) {
+    var animations = gltf.animations;
+    var nodes = gltf.nodes;
+
+    //Give each node an extra property to check if they are not targeted for animation.
+    for (var animationId in animations) {
+        if (animations.hasOwnProperty(animationId)) {
+            var animation = animations[animationId];
+            if (defined(animation.channels)) {
+                var channelLength = animation.channels.length;
+                for (var i = 0; i < channelLength; i++) {
+                    nodes[animation.channels[i].target.id].extras._pipeline.isMergeable = false;
+                }
+            }
+        }
+    }
+    for (var nodeId in nodes) {

```

Comment:
Why not put this loop first so we don't need the `if (!defined(node.extras._pipeline.isMergeable)) {` check?